packages/block-editor/src/components/block-list/block.js

```diff
@@ -511,22 +526,18 @@ function BlockListBlock( {
 				rootClientId={ rootClientId }
 			/>
 			{ isFirstMultiSelected && (
-				<BlockMultiControls rootClientId={ rootClientId } />
+				<BlockMultiControls
+					rootClientId={ rootClientId }
+					moverDirection={ moverDirection }
+				/>
 			) }
-			<div className="editor-block-list__block-edit block-editor-block-list__block-edit">
-				{ shouldRenderMovers && (
-					<BlockMover
-						clientIds={ clientId }
-						blockElementId={ blockElementId }
-						isHidden={ ! isSelected }
-						isDraggable={
-							isDraggable !== false &&
-							( ! isPartOfMultiSelection && isMovable )
-						}
-						onDragStart={ onDragStart }
-						onDragEnd={ onDragEnd }
-					/>
+			<div
+				className={ classnames(
+					'editor-block-list__block-edit block-editor-block-list__block-edit',
+					{ 'has-mover-inside': moverDirection === 'horizontal' },
 				) }
+			>
+				{ shouldRenderMovers && ( moverDirection === 'vertical' ) && blockMover }
```

Comment:
nit: This reads like it's a condition with no body. I'd use the ternary `?:` statement to reduce ambiguity.

```suggestion
				{ shouldRenderMovers && moverDirection === 'vertical' ? blockMover : null }
```